superset-frontend/src/utils/getClientErrorObject.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { SupersetClientResponse } from '@superset-ui/connection';
+import { t } from '@superset-ui/translation';
+import COMMON_ERR_MESSAGES from './errorMessages';
+
+// The response always contains an error attribute, can contain anything from the
+// SupersetClientResponse object, and can contain a spread JSON blob
+export type ClientErrorObject = {
+  error: string;
+  severity?: string;
+  message?: string;
```

Comment:
what kind of information that `error` and `message` are supposed to provide?

Reply:
error is the current error message that we display everywhere, message is expected in `setupApp` right now.

note that none of this is new, it's simply explicitly defining the possible responses from this function currently. In future prs I'll be standarizing this further